core/src/components/modal/test/animations/modal.e2e.ts

```diff
@@ -0,0 +1,45 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ config, title }) => {
+  test.describe(title('modal: animations'), () => {
+    test.beforeEach(async ({ page }) => {
+      await page.setContent(
+        `
+        <ion-modal is-open="true" trigger="open-modal"></ion-modal>
+      `,
+        config
+      );
+    });
+    test('card modal should clean up animations on dismiss', async ({ page }, testInfo) => {
+      testInfo.annotations.push({
+        type: 'issue',
+        description: 'https://github.com/ionic-team/ionic-framework/issues/28352',
+      });
+
+      const ionModalDidDismiss = await page.spyOnEvent('ionModalDidDismiss');
+
+      const modal = page.locator('ion-modal');
+
+      const initialAnimations = await page.evaluate(() => {
+        return document.querySelector('ion-modal')!.shadowRoot!.getAnimations();
+      });
+
+      // While the modal is open, it should have animations
+      await expect(initialAnimations.length).toBeGreaterThan(0);
+
+      await modal.evaluate((el: HTMLIonModalElement) => {
+        el.dismiss();
+      });
+
+      await ionModalDidDismiss.next();
+
+      const currentAnimations = await page.evaluate(() => {
+        return document.querySelector('ion-modal')!.shadowRoot!.getAnimations();
+      });
```

Comment:
Nit: You can use the existing locator to run your evaluation:

```suggestion
      const currentAnimations = await modal.evaluate((el: HTMLIonModalElement) => {
        return el.shadowRoot!.getAnimations();
      });
```